contracts/src/ethereum/TokenBridge.sol

```diff
@@ -0,0 +1,164 @@
+        _;
+    }
+
+    /*
+      Gets the addresses of bridgedToken & messagingContract from the ProxySupport initialize(),
+      and sets the storage slot accordingly.
+    */
+    function initializeContractState(bytes calldata data) internal override {
+        (uint256 l2TokenBridge_, StarknetMessaging messagingContract_) = abi.decode(
+            data,
+            (uint256, StarknetMessaging)
+        );
+
+        require(both(l2TokenBridge_ != 0, l2TokenBridge_ < CairoConstants.FIELD_PRIME), "L2_ADDRESS_OUT_OF_RANGE");
+
+        messagingContract = messagingContract_;
+        l2TokenBridge = l2TokenBridge_;
+    }
+
+    // TODO: implement proper encoding on js side
+    // function initializeWithoutProxy(bytes calldata data) public {
+    function initializeWithoutProxy(uint256 l2TokenBridge_, StarknetMessaging messagingContract_) public {
+        // following code initialize the state like`initialize()` function of ProxySupport contract
+        // validateInitData(data);
+        initGovernance();
+
+        // this part corresponds to `initializeContractState()` function of StarknetTokenBridge contract
+        // (uint256 l2TokenBridge_, StarknetMessaging messagingContract_) = abi.decode(
+        //     data,
+        //     (uint256, StarknetMessaging)
+        // );
+        require(both(l2TokenBridge_ != 0, l2TokenBridge_ < CairoConstants.FIELD_PRIME), "L2_ADDRESS_OUT_OF_RANGE");
+        messagingContract = messagingContract_;
+        l2TokenBridge = l2TokenBridge_;
+    }
+
+
+    // The selector of the "handle_deposit" l1_handler on L2.
+    // TODO: Recompute hash when the signature is decided
+    uint256 constant DEPOSIT_HANDLER =
+        1285101517810983806491589552491143496277809242732141897358598292095611420389;
+    uint256 constant TRANSFER_FROM_STARKNET = 0;
+
+    modifier isValidL2Address(uint256 l2Address) {
+        require(both(l2Address != 0, l2Address < CairoConstants.FIELD_PRIME), "L2_ADDRESS_OUT_OF_RANGE");
+        _;
+    }
+
+    function approveBridge(address l1Token, uint256 l2Token)
+        external
+        onlyGovernance
+        isValidL2Address(l2Token)
+    {
+        require(l1Token != address(0x0), "l1Token address cannot be 0x0");
+
+        uint256 l2Token_ = l1TokentoL2Token[l1Token];
+        require(l2Token_ == 0, "l2Token already set");
+
+        emit LogBridgeAdded(l1Token, l2Token);
+        l1TokentoL2Token[l1Token] = l2Token;
+    }
+
+    function sendMessage(address l1Token, uint256 l2Recipient, uint256 amount)
+        internal
+        isApprovedToken(l1Token)
+        isValidL2Address(l2Recipient)
+    {
+        emit LogDeposit(msg.sender, l1Token, amount, l2Recipient);
+
+        uint256 l2TokenAddress = l1TokentoL2Token[l1Token];
+
+        uint256[] memory payload = new uint256[](4);
+        payload[0] = l2Recipient;
+        payload[1] = l2TokenAddress;
```

Comment:
@swapnilraj Here, I use only one slot for `l2TokenAddress` instead of 2 previously, since we use only one slot for `l2Recipient`. We used to reconstruct the address on L2 side then, I've also removed that.

Reply:
yea that seems right